SwiftWisdom/Core/Downloader/Downloader.swift

```diff
@@ -9,57 +9,66 @@
 import UIKit
 
 public final class Downloader : NSObject, NSURLSessionDownloadDelegate {
-    
+
     public enum State {
         case Began
         case InProgress(CGFloat)
         case Completed(Result<NSURL>)
     }
-    
+
+    public enum Error : ErrorType {
```

Comment:
Just a heads up here - ErrorType is renamed Error in Swift 3. I don't know how we intend to handle this, but for now you can side step by renaming this DownloaderError. I know, super ugly :(